src/speculators/train/logger.py

```
@@ -479,6 +479,119 @@ def __init__(
+                )
+        if self.client.active_run() is None or self._run_id:
+            if self._experiment_name:
+                self.client.set_experiment(self._experiment_name)
+            self._run = self.client.start_run(run_name=self.init_kwargs["name"], nested=False)
+            self._auto_end_run = True
+        else:
+            self._run = self.client.active_run()
+
+        mlflow_tags = os.getenv("MLFLOW_TAGS", None)
+        if mlflow_tags:
+            mlflow_tags = json.loads(mlflow_tags)
```

Ruff (0.15.15)

[error] Undefined name `json`

(F821)